File: andriodGL/app/src/main/java/z3d/frame3d/FrameLinePointVo.java
package z3d.frame3d;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;

import z3d.base.Object3D;

public class FrameLinePointVo extends Object3D {
    public float time;
    public int id;
    public boolean iskeyFrame;
    public boolean isAnimation;
    public Object data;

    public static int maxTime = 0;

    public void writeObject(JSONObject $obj) {

        try {

            this.time = $obj.getInt("time");
//            this.id = $obj.get("id")!=null?(int) $obj.get("id"):0;
            this.iskeyFrame = $obj.getBoolean("iskeyFrame");
            this.isAnimation = $obj.getBoolean("isAnimation");


            this.x = $obj.getLong("x") / 10;
            this.y =  $obj.getLong("y")  / 10;
            this.z = $obj.getLong("z")  / 10;

            this.scaleX =  $obj.getLong("scaleX") / 10;
            this.scaleY =  $obj.getLong("scaleY")  / 10;
            this.scaleZ =  $obj.getLong("scaleZ")  / 10;

            this.rotationX =  $obj.getLong("rotationX") ;
            this.rotationY =  $obj.getLong("rotationY") ;
            this.rotationZ = $obj.getLong("rotationZ") ;

            this.data = $obj.get("data");


            FrameLinePointVo.maxTime = Math.max((int)this.time, FrameLinePointVo.maxTime);

        } catch (Exception e) {
            e.printStackTrace();
        }




    }
}

File: h5gl/src/Pan3d/frame3d/FrameLinePointVo.ts
module Pan3d {
    export class FrameLinePointVo extends Object3D {

        public time: number;
        public id: number;
        public iskeyFrame: boolean;
        public isAnimation: boolean;
        public data: any

        public static maxTime: number = 0
        public writeObject($obj: any): void {

            this.time = $obj.time;
            this.id = $obj.id;
            this.iskeyFrame = $obj.iskeyFrame;
            this.isAnimation = $obj.isAnimation;



            this.x = $obj.x / 10;
            this.y = $obj.y / 10;
            this.z = $obj.z / 10;

            this.scaleX = $obj.scaleX / 10;
            this.scaleY = $obj.scaleY / 10;
            this.scaleZ = $obj.scaleZ / 10;

            this.rotationX = $obj.rotationX;
            this.rotationY = $obj.rotationY;
            this.rotationZ = $obj.rotationZ;

            this.data = $obj.data;

            FrameLinePointVo.maxTime = Math.max(this.time, FrameLinePointVo.maxTime)

        }
    }

}